src/boundaries/BoxContainer.cpp

```diff
@@ -0,0 +1,23 @@
+#include <cmath>
+
+#include "BoxContainer.h"
+
+BoxContainer::BoxContainer(const double rc, const std::array<double, 3>& new_domain)
+    : ParticleContainer(new_domain) {
+    cells = CellList(rc, std::ceil(domain[0] / rc), std::ceil(domain[1] / rc), std::ceil(domain[2] / rc));
+    cells.create_list(particles);
+};
+
+BoxContainer::BoxContainer(const std::vector<Particle>& new_particles, const double rc, const std::array<double, 3>& new_domain)
+    : ParticleContainer(new_particles, new_domain) {
+    cells = CellList(rc, std::ceil(domain[0] / rc), std::ceil(domain[1] / rc), std::ceil(domain[2] / rc));
```

Comment:
you have gone for creating smaller cells, which I find a bit problematic because now just iterating over neighboring cells no longer gives you all the interaction in the cut off. Using `std::floor` and making cells slightly bigger would have been a better approach.